packages/markdown-it-plugins/src/util.ts

```diff
@@ -0,0 +1,78 @@
+import { ojs2notebook } from "@hpcc-js/observablehq-compiler";
+import { RenderNode } from "./render.ts";
+
+export interface FenceInfo {
+    type: "js" | "javascript" | string;
+    exec?: boolean;
+    echo?: boolean;
+    hide?: boolean;
+}
+
+export const fenceInfoDefaults: Readonly<FenceInfo> = {
+    type: "js",
+    exec: false,
+    echo: undefined,
+    hide: undefined
+};
+
+export function showSrc(fetchInfo: FenceInfo): boolean {
+    if (fetchInfo.exec === true) {
+        return fetchInfo.echo === true;
+    }
+    return fetchInfo.echo !== false;
+}
+
+export function executeSrc(fenceInfo: FenceInfo): boolean {
+    return fenceInfo.exec === true;
+}
+
+export function renderExecutedSrc(fetchInfo: FenceInfo): boolean {
+    return fetchInfo.exec == true && fetchInfo.hide !== true;
+}
+
+let idx = 0;
+function calcPlaceholders(content: string, fenceInfo: FenceInfo): RenderNode[] {
+    const retVal: RenderNode[] = [];
+    ++idx;
+    try {
+        const cellNb = ojs2notebook(content);
+        for (let i = 0; i < cellNb.nodes.length; ++i) {
```

Comment:
this looks like it always steps over the first node, is that correct?

Reply:
It shouldn't, both `for (let i = 0; i < cellNb.nodes.length; ++i)` and `for (let i = 0; i < cellNb.nodes.length; i++)` are the same thing, FWIW in old c++ compilers the ++i was was more efficient, so its just a habit.

(assuming that is what your picking up on)?